src/components/sign/SignIndex.js

```diff
@@ -0,0 +1,39 @@
+import { useState, useContext } from "react";
+import { Link } from "react-router-dom";
+import MainContext from "../../contexts/MainContext";
+import '../../styles/sign.css'
+import SignIn from "./SignIn";
+import SignUp from "./SignUp";
+
+const SignIndex = () => {
+    const [signView, setSignview] = useState(false);
```

Comment:
since this is indecating whether the ```signup``` component is visible or not I believe it'll be better to name it ```signupView``` and ```setSignupView``` instead